docs/upgrade.md

```diff
@@ -3,5 +3,263 @@ title: Upgrade guide
 description: Guide to update between major Powertools for AWS Lambda (TypeScript) versions
 ---
 
-!!! warning
-    This guide is a work in progress. We'll update it as we get closer to the 2.0 release. If you have any questions, or want to follow the progress, please join the discussion on the [GitHub issue](https://github.com/aws-powertools/powertools-lambda-typescript/issues/1714).
\ No newline at end of file
+## Migrate from v1 to v2
+
+To make your transition to v2 as smooth as possible while still providing you with the most up-to-date features and patterns we have made targeted breaking changes to some of the utilities. This guide will help you migrate your existing codebase to v2.
+
+### Quick summary
+
+Below a list of the main changes introduced in v2, review the sections below for more details and code examples for the changes that apply to your codebase.
```

Comment:
```suggestion

V2 is focused on official support for ESM (ECMAScript modules). We've made other minimal breaking changes to make your transition to v2 as smooth as possible.

### Quick summary

```

This doesn't say much, if I'm honest. Suggestion: tell it what it is upfront, then present the quick summary.